components/filter/filter.js

```diff
@@ -0,0 +1,259 @@
+	}
+
+	static get styles() {
+		return [bodyStandardStyles, css`
+			div[slot="header"] {
+				padding: 18px 6px;
+			}
+			.d2l-filter-dimension-header {
+				align-items: center;
+				display: flex;
+				width: 100%;
+			}
+			.d2l-filter-dimension-header-text {
+				padding-right: calc(2rem + 2px);
+				text-align: center;
+				width: 100%;
+			}
+			:host([dir="rtl"]) .d2l-filter-dimension-header-text {
+				padding-left: calc(2rem + 2px);
+				padding-right: 0;
+			}
+
+			.d2l-filter-dimension-value-text {
+				overflow: hidden;
+				text-overflow: ellipsis;
+				white-space: nowrap;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.disabled = false;
+		this._dimensions = new Map();
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this._dimensionsSlot = this.shadowRoot.querySelector('slot');
+
+		// Prevent these events from bubbling out of the filter
+		this.addEventListener('d2l-hierarchical-view-hide-complete', this._stopPropagation);
+		this.addEventListener('d2l-hierarchical-view-hide-start', this._stopPropagation);
+		this.addEventListener('d2l-hierarchical-view-show-complete', this._stopPropagation);
+		this.addEventListener('d2l-hierarchical-view-show-start', this._stopPropagation);
+		this.addEventListener('d2l-hierarchical-view-resize', this._stopPropagation);
+	}
+
+	render() {
+		let header = null;
+		if (this._activeDimensionKey) {
+			const dimensionText = this._dimensions.get(this._activeDimensionKey).text;
+			header = html`
+				<div class="d2l-filter-dimension-header">
+					<d2l-button-icon
+						@click="${this._handleDimensionHide}"
+						icon="tier1:chevron-left"
+						text="${this.localize('components.menu-item-return.returnCurrentlyShowing', 'menuTitle', dimensionText)}">
+					</d2l-button-icon>
+					<div class="d2l-filter-dimension-header-text d2l-body-standard">${dimensionText}</div>
+				</div>
+			`;
+		}
+
+		const dimensions = this._buildDimensions();
+
+		return html`
+			<d2l-dropdown-button-subtle
+				@d2l-dropdown-close="${this._handleDropdownClose}"
+				@d2l-dropdown-open="${this._stopPropagation}"
+				@d2l-dropdown-position="${this._stopPropagation}"
+				text="${this.localize('components.filter.filters')}"
+				?disabled="${this.disabled}">
+				<d2l-dropdown-menu min-width="285" max-width="420" no-padding-header>
+					<div slot="header">${header}</div>
+					<d2l-menu label="${this.localize('components.filter.filters')}">
+						<slot
+							@d2l-filter-dimension-data-change="${this._handleDataChangeListDimension}"
+							@d2l-filter-dimension-value-data-change="${this._handleDataChangeListDimensionValue}"
+							@d2l-filter-dimension-slot-change="${this._handleSlotChange}"
+							@slotchange="${this._handleSlotChange}"
+						></slot>
+						${dimensions}
+					</d2l-menu>
+				</d2l-dropdown-menu>
+			</d2l-dropdown-button-subtle>
+		`;
+	}
+
+	_buildDimension(dimension) {
+		let dimensionHTML;
+		switch (dimension.tagName.toLowerCase()) {
```

Comment:
Setting things up here to eventually have more than one type of dimension.  A lot of function names have been chosen to group things nicely once that happens as well.